client/client_test.go

```diff
@@ -0,0 +1,1683 @@
+					c := NewWrappedReader(k8sAPIReader, scheme)
+					err = c.Get(testCtx, types.NamespacedName{Name: lv.Name}, checklv)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(checklv.Name).Should(Equal(fmt.Sprintf("current-%d", i)))
+					Expect(checklv.Spec.Name).Should(Equal(fmt.Sprintf("current-%d", i)))
+					Expect(checklv.Spec.NodeName).Should(Equal(fmt.Sprintf("node-%d", i)))
+					Expect(checklv.Spec.DeviceClass).Should(Equal(topolvm.DefaultDeviceClassName))
+					Expect(checklv.Spec.Size.Value()).Should(Equal(resource.NewQuantity(1<<30, resource.BinarySI).Value()))
+					Expect(checklv.Spec.Source).Should(Equal(fmt.Sprintf("source-%d", i)))
+					Expect(checklv.Spec.AccessType).Should(Equal("rw"))
+					Expect(checklv.Status.VolumeID).Should(Equal(fmt.Sprintf("volume-%d", i)))
+					Expect(checklv.Status.Code).Should(Equal(codes.Unknown))
+					Expect(checklv.Status.Message).Should(Equal(codes.Unknown.String()))
+					Expect(checklv.Status.CurrentSize.Value()).Should(Equal(resource.NewQuantity(1<<30, resource.BinarySI).Value()))
+				})
+			})
+
+			Context("List", func() {
+				It("standard case", func() {
+					for i := 0; i < 2; i++ {
+						lv := currentLV(i)
+						err := k8sDelegatedClient.Create(testCtx, lv)
+						Expect(err).ShouldNot(HaveOccurred())
+
+						setCurrentLVStatus(lv, i)
+						err = k8sDelegatedClient.Status().Update(testCtx, lv)
+						Expect(err).ShouldNot(HaveOccurred())
+					}
+
+					lvlist := new(topolvmv1.LogicalVolumeList)
+					c := NewWrappedReader(k8sAPIReader, scheme)
+					err := c.List(testCtx, lvlist)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(len(lvlist.Items)).Should(Equal(2))
+					for i, lv := range lvlist.Items {
+						Expect(lv.Name).Should(Equal(fmt.Sprintf("current-%d", i)))
+						Expect(lv.Spec.Name).Should(Equal(fmt.Sprintf("current-%d", i)))
+						Expect(lv.Spec.NodeName).Should(Equal(fmt.Sprintf("node-%d", i)))
+						Expect(lv.Spec.DeviceClass).Should(Equal(topolvm.DefaultDeviceClassName))
+						Expect(lv.Spec.Size.Value()).Should(Equal(resource.NewQuantity(1<<30, resource.BinarySI).Value()))
+						Expect(lv.Spec.Source).Should(Equal(fmt.Sprintf("source-%d", i)))
+						Expect(lv.Spec.AccessType).Should(Equal("rw"))
+						Expect(lv.Status.VolumeID).Should(Equal(fmt.Sprintf("volume-%d", i)))
+						Expect(lv.Status.Code).Should(Equal(codes.Unknown))
+						Expect(lv.Status.Message).Should(Equal(codes.Unknown.String()))
+						Expect(lv.Status.CurrentSize.Value()).Should(Equal(resource.NewQuantity(1<<30, resource.BinarySI).Value()))
+					}
+				})
+			})
+		})
+
+		Context("wrappedClient", func() {
+			Context("Get", func() {
+				It("standard case", func() {
+					i := 0
+					lv := currentLV(i)
+					err := k8sDelegatedClient.Create(testCtx, lv)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					setCurrentLVStatus(lv, i)
+					err = k8sDelegatedClient.Status().Update(testCtx, lv)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					Eventually(func(g Gomega) {
```

Comment:
Because the wrappedClient uses delegated client in internal in this test case.
the Get method of delegated client is not synchronous because delegated client uses cache(indexer).